spec/controllers/two_factor_authentication/piv_cac_verification_controller_spec.rb

```diff
@@ -98,6 +110,34 @@
       it 'displays flash error message' do
         expect(flash[:error]).to eq t('devise.two_factor_authentication.invalid_piv_cac')
       end
+
+      it 'resets the piv/cac session information' do
+        expect(subject.user_session[:decrypted_x509]).to be_nil
+      end
+    end
+
+    context 'when the user presents a different piv/cac' do
+      before do
+        sign_in_before_2fa(user)
+
+        get :show, params: { token: 'good-other-token' }
+      end
+
+      it 'increments second_factor_attempts_count' do
+        expect(subject.current_user.reload.second_factor_attempts_count).to eq 1
+      end
+
+      it 're-renders the piv/cac entry screen' do
+        expect(response).to render_template(:show)
+      end
+
+      it 'displays flash error message' do
+        expect(flash[:error]).to eq t('devise.two_factor_authentication.invalid_piv_cac')
+      end
+
+      it 'resets the piv/cac session information' do
+        expect(subject.user_session[:decrypted_x509]).to be_nil
+      end
```

Comment:
Our preference for controller specs is to use the stubbing methods, such as `stub_sign_in_before_2fa`. If that's not possible, such as when needing to make DB changes, then our convention is to keep all expectations in the same `it` block to minimize the amount of DB calls and to avoid slowing down the tests.

Reply:
Cool. I'll take a look at the subbed method and see what I can simplify here.